com.unity.sg2/Editor/GraphUI/GraphElements/Views/ShaderGraphViewSelection.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Linq;
+using UnityEditor.GraphToolsFoundation.Overdrive;
+using UnityEditor.GraphToolsFoundation.Overdrive.BasicModel;
+using UnityEngine.UIElements;
+
+namespace UnityEditor.ShaderGraph.GraphUI
+{
+    public class ShaderGraphViewSelection : GraphViewSelection
+    {
+        public ShaderGraphViewSelection(GraphView view, GraphModelStateComponent graphModelState, SelectionStateComponent selectionState)
+            : base(view, graphModelState, selectionState)
+        {
+
+        }
+
+        /// <summary>
+        /// Adds items related to the selection to the contextual menu.
+        /// </summary>
+        /// <param name="evt">The contextual menu event.</param>
+        public override void BuildContextualMenu(ContextualMenuPopulateEvent evt)
+        {
+            if (evt.menu.MenuItems().Count > 0)
+                evt.menu.AppendSeparator();
+
+            if (GetSelection().Any(model => model is INodeModel))
+                HandleNodeContextMenus(evt);
+            else
+                HandleEdgeContextMenu(evt);
+        }
+
+        void HandleEdgeContextMenu(ContextualMenuPopulateEvent evt)
+        {
+            evt.menu.AppendAction("Delete", _ =>
```

Comment:
In Old SG, edges only have two context menu options, "Delete" and "Add Redirect Nodes"